src/lib/FileHelper.ts

```diff
@@ -92,9 +97,11 @@ export function findFiles(include: RelativePattern, maxResults?: number, errIfNo
 	return new Observable<Uri>(observer => {
 		const tokenSource = new CancellationTokenSource();
 
+		let ignoreList: string[] = getIgnoreList();
+		
 		workspace.findFiles(
 			include,
-			undefined,
+			new RelativePattern(workspace.workspaceFolders![0], '**/' + ignoreList[0]),
```

Comment:
@SqrTT I don't really like this.
Maybe I find a better solution by tomorrow.

I would like to see all directories from the ignoreList beeing included here.
In addition I think this is not safe to assume node_modules stays first in the array.

If you have a cool solution for this, I'd be happy :)

Reply:
Used glob pattern to exclude files

Reply:
not sure if excluding node_modules solves the issue.

Actual issue is with duplication. Cartridges may duplicate  within workspaces.

Reply:
I tested the issue and was able to resolve it using this approach.

  since node_modules are set to the ignorelist therefore we should also exclude the cartridges, what do you think?

Reply:
correct, this resolve this issue, but bring another where users can't use cartridges form npm. And doesn't solve case when the same cartridge exists in different cartridges.

Reply:
Ok I was thinking that the following line will in general restrict `node_modules` folder
```
workspace.getConfiguration('extension.prophet').get('ignore.list', ['node_modules', '\\.git', '\\.zip$']);
```
But it looks like I miss interpreted this - so it is supported to have cartridges in `node_modules` ?